lib/rspec/core/pending.rb

```diff
@@ -51,36 +51,12 @@ class PendingExampleFixedError < StandardError; end
       #         raise "broken"
       #       end
       #     end
-      #
-      # @note `before(:example)` hooks are eval'd when you use the `pending`
-      #   method within an example. If you want to declare an example `pending`
-      #   and bypass the `before` hooks as well, you can pass `:pending => true`
-      #   to the `it` method:
-      #
-      #       it "does something", :pending => true do
-      #         # ...
-      #       end
-      #
-      #   or pass `:pending => "something else getting finished"` to add a
-      #   message to the summary report:
-      #
-      #       it "does something", :pending => "something else getting finished" do
-      #         # ...
-      #       end
```

Comment:
This could be reworded rather than removed, perhaps mentioning that only the failures in the block cause the behaviour?

Reply:
The reason I've removed this was that `If you want to declare an example `pending` and bypass the `before` hooks as well` was misleading. Replaced it with `pending` example method docs.

Reply:
Sorry, I don't understand what needs to remain.
As far as I understand, hooks aren't skipped in either case. In this regards, there is no alternative.
And all of:
```ruby
pending 'example method' do
  # something that fails
end
it 'has a cool feature', pending: 'yet to be implemented' do # or just pending: true
  # cool missing feature test
end
it 'should work' do
  pending 'but does not'
  # and a failing expectation
end
```
are just different notations of the same example group that will pass if its expectations fail.

Am I missing something? I'm all for clarifying/extending the docs, but at this point I'm out of ideas on how to do this exactly.

Reply:
This doc is explaining this difference:
```
RSpec.describe do
  before do
    raise "BOOM!"
  end

  it "will pend", pending: "Oh yes" do
    raise "Yeah nah"
  end

  it "will fail" do
    pending do
      raise "Won't happen"
    end
  end
end
```

```
*F

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) will pend
     # Oh yes
     Failure/Error: raise "BOOM!"

     RuntimeError:
       BOOM!
     # ./spec/pending_example_spec.rb:3:in `block (2 levels) in <top (required)>'

Failures:

  1) will fail
     Failure/Error: raise "BOOM!"

     RuntimeError:
       BOOM!
     # ./spec/pending_example_spec.rb:3:in `block (2 levels) in <top (required)>'

Finished in 0.00349 seconds (files took 0.13413 seconds to load)
2 examples, 1 failure, 1 pending

Failed examples:

rspec ./spec/pending_example_spec.rb:10 # will fail


```

Reply:
The difference being that pending metadata will cause a hook failure to be considered as part of the pending nature of the spec, where as the block will cause a failure as it hasn't reached the bit that can be pending.

Reply:
Got it, thanks for the clarification!
Let me quickly get this note back.

Reply:
Done.
I've extended the previous docs based on the assumption that if it wasn't clear at all for me, it wasn't clear for the majority.

I can extract those docs changes into separate PR, one for `4-0-dev`, and another for `main` if you like.
Surely it doesn't relate to "Remove deprecations" much, especially in the part of `pending.rb` where only the error message was reduced not to mention RSpec 2.

Reply:
#2872 #2873